reconcile/utils/imap_client.py

```diff
@@ -0,0 +1,54 @@
+import imaplib
+from typing import Union
+
+from reconcile.utils.secret_reader import SecretReader
+
+
+class ImapClient:
+    def __init__(self, settings):
+        imap_config = self.get_imap_config(settings)
+        self.host: str = imap_config["server"]
+        self.user: str = imap_config["username"]
+        self.password: str = imap_config["password"]
+        self.port: int = int(imap_config["port"])
+        self.timeout: int = settings["imap"].get("timeout", 30)
+        self._server: Union[imaplib.IMAP4_SSL, None] = None
+
+    def __enter__(self):
+        self._server = imaplib.IMAP4_SSL(
+            host=self.host, port=self.port, timeout=self.timeout
+        )
+        self._server.login(self.user, self.password)
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        if self._server:
+            self._server.logout()
+
+    @property
+    def server(self):
```

Comment:
nit: consider making this "internal" because it is not used from outside of the class right now.